web/packages/teleport/src/services/api/api.ts

```diff
@@ -142,66 +142,31 @@ const api = {
     customOptions: RequestInit,
     mfaResponse?: MfaChallengeResponse
   ): Promise<any> {
-    const response = await api.fetch(url, customOptions, mfaResponse);
-
-    let json;
     try {
-      json = await response.json();
+      return await api.fetch(url, customOptions, mfaResponse);
     } catch (err) {
-      // error reading JSON
-      const message = response.ok
-        ? err.message
-        : `${response.status} - ${response.url}`;
-      throw new ApiError({ message, response, opts: { cause: err } });
-    }
-
-    if (response.ok) {
-      return json;
-    }
-
-    /** This error can occur in the edge case where a role in the user's certificate was deleted during their session. */
-    const isRoleNotFoundErr = isRoleNotFoundError(parseError(json));
-    if (isRoleNotFoundErr) {
-      websession.logoutWithoutSlo({
-        /* Don't remember location after login, since they may no longer have access to the page they were on. */
-        rememberLocation: false,
-        /* Show "access changed" notice on login page. */
-        withAccessChangedMessage: true,
-      });
-      return;
+      // Retry with MFA if we get an admin action MFA error.
+      if (!mfaResponse && isAdminActionRequiresMfaError(err)) {
+        mfaResponse = await api.getAdminActionMfaResponse();
+        return await api.fetch(url, customOptions, mfaResponse);
```

Comment:
No need for the `await`s here

Reply:
Yeah, it's not a big deal, but if you have a promise, you can return it from an async function without awaiting.

I'm not sure if there's a lint rule which would automatically take care of this. await-thenable and no-misused-promises seem to be about different cases entirely. https://github.com/typescript-eslint/typescript-eslint/issues/8517

Reply:
I see, I wasn't sure how the try/catch would work without the first await

Reply:
Sorry @Joerger, you're absolutely correct

<img width="802" alt="image" src="https://github.com/user-attachments/assets/fde636c2-660f-4334-b34c-45ecea633856" />

Reply:
No problem, reverted that one.